File: app/src/main/java/com/aiden/andmodule/model/MainViewModel.java
package com.aiden.andmodule.model;
import android.app.Application;
import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.room.Room;
import com.aiden.andmodule.db.LiveAppDatabase;
import com.aiden.andmodule.db.LiveTodo;

import java.util.List;

public class MainViewModel extends AndroidViewModel {
    private LiveAppDatabase db;
   public MainViewModel(@NonNull Application application) {
        super(application);
        db= Room.databaseBuilder(application, LiveAppDatabase.class,"todo-db")
                .allowMainThreadQueries()
                .build();
    }
    public LiveData<List<LiveTodo>> getAll(){
        return  db.todoDao().getAll();
    }
    public void insert(LiveTodo todo){
        db.todoDao().insert(new LiveTodo(todo.getTitle()));
    }
}

File: app/src/main/java/com/aiden/andmodule/livedata/MainViewModel.kt
package com.aiden.andmodule.livedata


import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

/**
 * Created by admin on 2017-08-03.
 */
class MainViewModel : ViewModel(){

    var isSwitch : MutableLiveData<Boolean> = MutableLiveData()

    var counter = MutableLiveData<Int>()


    fun increase() {
        counter.value = this.counter.value!! + 1
    }








}